### ISSUE TITLE
Investigate code coverage issues

### ISSUE THREAD

**Issue by MysterAitch** [javaparser/javaparser#3190]
Codecov is reporting some methods as being not covered (therefore indicating 0% patch coverage), but running the tests within the debugger with breakpoints on the relevant line shows that it is executed.

More info / example within #3187

> I've just had a closer look and running it through a debugger with breakpoints in each method, both of them within `ResolvedAnnotationDeclaration` gets run.
> 
> It might be an issue with codecov or perhaps with the maven surefire plugin. I'll open up a separate issue and get this one merged in :+1:
>
>_Originally posted by @MysterAitch in https://github.com/javaparser/javaparser/issues/3187#issuecomment-805183676_

**Comment by MysterAitch** [comment]
Some screenshots of the codecov "0%" patch coverage

![image](https://user-images.githubusercontent.com/1537214/112208988-e2f4ee00-8c10-11eb-8667-fc0a5df39d4a.png)

![image](https://user-images.githubusercontent.com/1537214/112209096-0029bc80-8c11-11eb-91d5-6150ee61ce49.png)

---

![image](https://user-images.githubusercontent.com/1537214/112209289-3404e200-8c11-11eb-8865-3c5e9c30f0f0.png)

---

![image](https://user-images.githubusercontent.com/1537214/112208801-ade89b80-8c10-11eb-8c4d-7a6e812cecfc.png)

![image](https://user-images.githubusercontent.com/1537214/112208854-bb058a80-8c10-11eb-9201-ff33b62dcf8b.png)

![image](https://user-images.githubusercontent.com/1537214/112208864-be991180-8c10-11eb-9eee-d14a5b8f661a.png)

**Comment by MysterAitch** [comment]
It seems this might be due to codecov not picking up on symbol solver coverage result (hence symbol solver code always having 0% coverage).

**Comment by MysterAitch** [comment]
Update - definitely the case.

Looking through recent action runs, codecov isn't picking up the jacoco results for the symbol solver tests, but it does for the core tests (screenshots below).

Just need to figure out if this is because the file isn't created/written to, or if codecov isn't finding it.


![image](https://user-images.githubusercontent.com/1537214/112548721-934f2780-8db4-11eb-897b-144514485985.png)

![image](https://user-images.githubusercontent.com/1537214/112548493-3bb0bc00-8db4-11eb-967f-33f43538978e.png)

**Comment by MysterAitch** [comment]
Multiple PRs here: 

#3199, #3201, #3203, #3208

**Comment by MysterAitch** [comment]
The change within #3199 (and then re-implemented within #3208) is confirmed to now allow the jacoco test / coverage reports be produced. Without this, the file `jacoco.xml` fails to be produced.

Manually checking the reports via codecov.io shows that it now picks these up, and the code coverage increases accordingly.

**Comment by MysterAitch** [comment]
**Here is a screenshot of an arbitrary "older" commit, showing only javaparser-core**

![image](https://user-images.githubusercontent.com/1537214/112722714-05358700-8f03-11eb-8ffc-740c77b0a2ea.png)

**Here is a screenshot of #3208 "in progress" - note that it picks up the symbol solver code / directories now:**

![image](https://user-images.githubusercontent.com/1537214/112722825-8b51cd80-8f03-11eb-8426-360bf40905c9.png)

**Comment by MysterAitch** [comment]
The problem which prompted the later changes/tests is that there are delays with having the 54 coverage reports to be received, merged together, AND THEN also update Github -- both the PR comment and the status check. 

Also seen at : https://github.com/javaparser/javaparser/pull/3208#issuecomment-808737905


![image](https://user-images.githubusercontent.com/1537214/112722777-5180c700-8f03-11eb-857e-9a4f6096dfcc.png)



IIRC this will self-resolve once new PRs are created based on this change.



**For reference, here is an "in-progress" screenshot of the "build" tab of a commit -- here it shows each of the uploaded reports.**
Tip: Searching (ctrl+F) for "JDK=" or "OS=" lets you figure out how many uploads there are.

![image](https://user-images.githubusercontent.com/1537214/112722962-32366980-8f04-11eb-9bb8-cab240b91ca0.png)

**Comment by MysterAitch** [comment]
Cross-post: https://github.com/javaparser/javaparser/pull/3208#issuecomment-808738241

> Confirmation that all 54 reports have been uploaded to, and processed by, codecov (specifically note the top-right corner, showing the number of in-page search results for "jdk=")
> 
> ![image](https://user-images.githubusercontent.com/1537214/112723274-89890980-8f05-11eb-929f-00e164c4c06a.png)

**Comment by MysterAitch** [comment]
I just posted to the codecov forums to see if they're able to assist:

~https://community.codecov.io/t/github-status-check-no-coverage-information-found-on-head-and-pr-comment-missing/2689~

~EDIT: Hmm, I've hit the spam filter! I shall give a little time and see what happens... As a new user, maybe my post must be manually approved?~
~![image](https://user-images.githubusercontent.com/1537214/112724256-63b23380-8f0a-11eb-979c-0e6c99c7f110.png)~


EDIT2: I'm no longer a new/untrusted user on there, and have posted a thread/comment here with details:

https://community.codecov.io/t/no-coverage-information-found-on-head-on-github-and-no-pr-comment/2694

**Comment by MysterAitch** [comment]
Per comment in #3208, the code coverage stuff is now being reported "fine", just the PR comments from the codecov app and the status checks to sort out.

This might self-resolve now that PR is merged and other (new?) PRs are created using this commit as a "base" (educated guess/speculation).

> Coverage is being recorded correctly now, as confirmed in this screenshot below.
> 
> There's an ongoing problem with the status checks and pr comments from codecov, but this might self-resolve once this is merged in.
> 
> ![image](https://user-images.githubusercontent.com/1537214/112731401-ed73f800-8f2e-11eb-827c-82fab36896bf.png)

**Comment by MysterAitch** [comment]
A dump of screenshots for future reference re: #3208 :

![image](https://user-images.githubusercontent.com/1537214/112732333-33ca5680-8f31-11eb-8610-3c7628c32a0c.png)

![image](https://user-images.githubusercontent.com/1537214/112732360-670ce580-8f31-11eb-9b63-d783e47cb0d4.png)

![image](https://user-images.githubusercontent.com/1537214/112732356-5f4d4100-8f31-11eb-8f2a-232a0c264f3f.png)


![image](https://user-images.githubusercontent.com/1537214/112732338-417fdc00-8f31-11eb-983f-8a302d5c8849.png)

![image](https://user-images.githubusercontent.com/1537214/112732387-96235700-8f31-11eb-862b-684ad1de7ef1.png)

![image](https://user-images.githubusercontent.com/1537214/112732404-ad624480-8f31-11eb-9589-199f164607e2.png)

**Comment by jlerbsc** [comment]
I am closing this issue since a pr was merged
